pyogrio/_io.pyx

```diff
@@ -2023,25 +2094,38 @@ cdef create_ogr_dataset_layer(
     cdef OGRSpatialReferenceH ogr_crs = NULL
     cdef OGRwkbGeometryType geometry_code
     cdef int layer_idx = -1
+    cdef VSIStatBufL st_buf
+
 
     path_b = path.encode('UTF-8')
     path_c = path_b
 
     driver_b = driver.encode('UTF-8')
     driver_c = driver_b
 
+    if is_vsi:
+        path_exists = VSIStatExL(path_c, &st_buf, VSI_STAT_EXISTS_FLAG) == 0
```

Comment:
Do you if this is actually needed for the current scope of this PR? Because right now you just created this memfile with a unique uuid, so it will never already exist?

Reply:
Correct; no longer needed, it was left over from trying to append (can add back in later as part of that)